public_html/wp-content/plugins/wcpt/wcpt-event/class-event-admin.php

```diff
@@ -694,6 +697,59 @@ public function print_admin_notices() {
 
 	}
 
+	/**
+	 * Prevent users from creating new WordCamp and Meetup posts on dashboard. In most of the cases, all posts should be created
+	 * thru the public application forms in order to get all needed information and to initiate the vetting process correctly.
+	 *
+	 * Expectation is made for users with administrator and deputy roles, as they need to create events manually from time to time.
+	 *
+	 * Used wp_insert_post_empty_content hook is run fo creation and updates, which is why post ID needs to be checked. The hook
+	 * short circuits creation of new post when truthy value is returned.
+	 *
+	 * @param  boolean $maybe_empty Whether the post should be considered "empty".
+	 * @param  array   $postarr      Array of post data.
+	 *
+	 * @return boolean              Whether the post should be considered "empty".
+	 */
+	public function maybe_prevent_creation_of_new_post( $maybe_empty, $postarr ) {
+		$post_type = $postarr['post_type'];
+
+		// Apply only for WordCamp and Meetup post types.
+		if ( $this->get_event_type() !== $post_type ) {
+			return $maybe_empty;
+		}
+
+		// The action hooked into is used also when updating posts, which all users should be able to do based on their caps.
+		if ( ! empty( $postarr['ID'] ) ) {
+			return $maybe_empty;
+		}
+
+		// Doing the checks only on dashboard ensures that other use cases like application forms do still work.
+		if ( ! is_admin() ) {
+			return $maybe_empty;
+		}
+
+		// Allow admins to create new posts in event CPT's.
```

Comment:
For roles checking we should use `current_user_can` against a capability, which can probably also wrap the check for deputy or higher too.

Reply:
the parenthesis below appear to be missing spacing to be consistent with the rest of the code.

Reply:
Wanted to separate the admin and deputy checks to make it easier to see for who we are allowing the creation.

There is `wordcamp_wrangle_wordcamps` capability which seems to affect WordCamps and Meetups and should be assigned to deputies and mentors. Though not sure if it's given for admins, so we probably still need that check.

In any case, replacing the `is_deputy` with capability check seems necessary. Looking at the origins of that function again, it seems to check against `$trusted_deputies` array, which sounds a lot like a list that does not contain every deputy.

Reply:
Looks like admins do have the `wordcamp_wrangle_wordcamps` capability, at least in the Docker env.